store/postgres/src/deployment_store.rs

```diff
@@ -1174,6 +1174,18 @@ impl DeploymentStore {
                     layout.count_query.as_str(),
                     count,
                 )?;
+
+                // Delete subgraph errors
+                deployment::revert_subgraph_errors(conn, &site.deployment, block_ptr_to.number)?;
+
+                // Clear `failed` column back to `false`
+                deployment::update_deployment_status(
```

Comment:
@leoyvens I tried to think a few scenarios where we shouldn't call this, but I think the failed block will always be the last one and we'll always want the deployment to be healthy after a rewind.

Do you have any worries/considerations here?

Reply:
There is the case of non-fatal errors, where the subgraph health would be `Unhealthy`. If this checked that the health is actually failed I think it would be functionally correct, but we need a test.

Also there are some code organization issues with this approach. `revert_subgraph_errors` is already called from `revert_metadata` so that's redundant. And the health update should be moved into `revert_subgraph_errors` imo.

Reply:
Thanks for the feedback! I've addressed it by stopping calling `revert_subgraph_errors` twice. And changing it to also update the `failed` and `status` columns in the case of both of them being `failed` :wink: 

Do you think we should change an integration test for this?

Reply:
Nice! A test would be great